### PR TITLE
[enocean] Implemented ESP2 protocol

### LINKED ISSUE #6512: [enocean] Support ESP2
EnOcean radio telegrams are transfered using ESP3 protocol. However if you want to connect your gateway directly with a RS485 bus (wired setup) you have to use ESP2 protocol. Furthermore older radio gateways or TCP/IP bridges like Thermokon STC-Bridge also just support ESP2 protocol.

### PR DESCRIPTION
**Pull request by fruggy83**
Implemented ESP2 protocol
Fixes #6512
Signed-off-by: Daniel Weber <uni@fruggy.de>

### CODE REVIEW

File: bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/Helper.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.Arrays;
+
+/**
+ *
+ * @author Daniel Weber - Initial contribution
+ */
+public class Helper {
+
+    public static byte[] concatAll(byte[] a, byte[]... rest) {
+        int totalLength = a.length;
```
Comment (KarstenB) [inline bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/Helper.java]: Rest can be null, even tough you control all use-cases, I would add a:
```
if (rest == null)
   return a;
```
Even before accessing a.length (that really should not ever be null)

File: bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/config/EnOceanBridgeConfig.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        private String version;
+
+        ESPVersion(String version) {
+            this.version = version;
+        }
+
+        public static ESPVersion getESPVersion(String espVersion) {
+            for (ESPVersion version : values()) {
+                if (version.version.equals(espVersion)) {
```
Comment (KarstenB) [inline bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/config/EnOceanBridgeConfig.java]: Maybe adding equalsIgnoreCase for added user-friendliness?

File: bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/handler/EnOceanBridgeHandler.java [hunk 3]
```diff
@@ -193,34 +195,54 @@ private synchronized void initTransceiver() {
...
             transceiver.StartReceiving(scheduler);
 
-            if ((boolean) c.get(RS485)) {
-                baseId = new byte[4];
+            if (c.rs485) {
+                if (c.rs485BaseId != null && !c.rs485BaseId.isEmpty()) {
+                    baseId = HexUtils.hexToBytes(c.rs485BaseId);
+                    if (baseId.length != 4) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                "RS485 BaseId has wrong format");
```
Comment (KarstenB) [inline bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/handler/EnOceanBridgeHandler.java]: Maybe add a little more explanation here. Like: 
```RS485 BaseId has the wrong format. It is expected to be an 8 digit hex code, for example 01000000```

**Comment by fruggy83** [comment]
@KarstenB thansk for the hints, everything should be fixed now

**Comment by KarstenB** [comment]
I read the code and did not find any major issues, and I also tested it in production and it works flawlessly.

**Comment by fruggy83** [comment]
@kaikreuzer, @J-N-K on which branch should I do the rebase? Master (I guess this is for 3.0.0 development) or 2.5.x?

**Comment by kaikreuzer** [comment]
@fruggy83 Is it only backward compatible changes that would make sense to ship in a 2.5.1 patch release? In any case, rebase it on master for now, we will probably soon have some further update on the process.

**Comment by fruggy83** [comment]
@kaikreuzer I rebased my PR on 2.5.x and fixed the conflicts

**Comment by fruggy83** [comment]
Fixed conflicts regarding copyright headers

**Comment by fruggy83** [comment]
@kaikreuzer friendly pinging. 
Branch is already based on 2.5.x. Just leave a message if I still have anything to do. Thanks a lot in advance.

**Pull request merged**